samples/net/lwm2m_client/overlay-dtls.conf

```diff
@@ -3,7 +3,7 @@ CONFIG_LWM2M_PEER_PORT=5684
 
 # Select Zephyr mbedtls
 CONFIG_MBEDTLS=y
-CONFIG_MBEDTLS_TLS_VERSION_1_2=y
+CONFIG_MBEDTLS_SSL_PROTO_TLS1_2=y
```

Comment:
If we want to have compatible symbol names, we should rather rename them usptream (justifying by the fact its more consistent with mbed TLS nomenclature).

Reply:
@rlubos: If you have the capacity to do this, then that would be great!

MBEDTLS_TLS_VERSION_1_2 is a zephyr-specific configuration, from what I understand. I haven't seen this configuration elsewhere

Reply:
We are leaving this for a later time. Reverted removal